django/apps/registry.py

```diff
@@ -255,6 +255,17 @@ def get_registered_model(self, app_label, model_name):
                 "Model '%s.%s' not registered." % (app_label, model_name))
         return model
 
+    @lru_cache.lru_cache(maxsize=None)
+    def get_swappable_setting(self, to_string):
+        # See if anything swapped/swappable matches
+        for model in self.get_models(include_swapped=True):
+            if model._meta.swapped:
+                if model._meta.swapped == to_string:
+                    return model._meta.swappable
+            if ("%s.%s" % (model._meta.app_label, model._meta.object_name)) == to_string and model._meta.swappable:
```

Comment:
I have a hard time following the logic here. You could at least fold the second condition into the first `if model._meta.swapped`, which gives:

    if model._meta.swapped:
        if model._meta.swapped == to_string:
            return model._meta.swappable
        if ("%s.%s" % (model._meta.app_label, model._meta.object_name)) == to_string:
            return model._meta.swappable

which can be further "simplified" to:

    if model._meta.swapped and (
            model._meta.swapped == to_string or
            "%s.%s" % (model._meta.app_label, model._meta.object_name)) == to_string):
        return model._meta.swappable

(but I'm not sure that's an improvement).

The last condition is case-sensitive. However, for better or worse, model names are case-insensitive in Django. The proper test would be:

    to_app_label, to_model_name = to_string.split('.')
    to_model_name = to_model_name.lower()
    # and later, test if:
    (to_app_label, to_model_name) == (model._meta.app_label, model._meta.model_name)

Even then the purpose of this method is unclear to me. Is it a test "swappable" or "swapped"? Is `get_swappable_setting` the most self-explanatory name for this method?